### ISSUE TITLE
Something different with URL generation between 1.1.3 and 1.1.4

### ISSUE THREAD

**Issue by robofant** [bednee/cooluri#82]
It's a bit complicated to explain. Since 1.1.4 there is an extra section in the URL, generated from the name of the root level page.

Let me make some examples.
before: http://wwww.example.org/
after: http://www.example.org/name-of-root-level-page.html

before: http://wwww.example.org/foo/bar.html
after: http://wwww.example.org/name-of-root-level-page/foo/bar.html

It is a TYPO3 7.6 project, I only upgraded CoolURI from 1.1.3 to 1.1.4 and did not change the config file. After downgrade to CoolURI 1.1.3 everything is fine again.

But now I have a TYPO3 8.7 project and cannot use CoolURI 1.1.3. How can I deal with this weird behaviour of CoolURI 1.1.4?

**Comment by bednee** [comment]
Hi,
how does your structure look like? In order to have rootpage without name, the stucture should look like
Root (regular page)
- Home (shortcut to root)

Jan

**Comment by robofant** [comment]
Yes, this is my structure. But the root page has a name (the same name as the home page) for some reason. I don't know the reason anymore. But this was never a problem with 1.1.3 and earlier.

**Comment by robofant** [comment]
Ok, I found the problem. Here is the fix:

`--- CoolUri.php.org	2017-04-09 11:17:56.000000000 +0200
+++ CoolUri.php	2018-03-15 00:48:08.147040700 +0100
@@ -443,7 +443,7 @@
         $db = &$GLOBALS['TYPO3_DB'];
         $temp = $db->exec_SELECTquery('COUNT(*) as num', 'sys_template',
             'deleted=0 AND hidden=0 AND pid=' . $pagId . ' AND root=1' . $enable);
-        $countCache[$pagId] = (int)$db->sql_fetch_assoc($temp);
+        $countCache[$pagId] = $db->sql_fetch_assoc($temp);
         return $countCache[$pagId];
     }
`

**Comment by robofant** [comment]
Ok, I found the problem. Here is the fix.

[CoolUri.php.diff.txt](https://github.com/bednee/cooluri/files/1813523/CoolUri.php.diff.txt)

**Comment by bednee** [comment]
I believe that your patch is not correct. Have a look at the commit I've just made - https://github.com/bednee/cooluri/commit/781866b9e238eb6b483bcea125bd9d0d7c563e57 - that's how it should be I believe. Please check it out and let me know. Thanks.

**Comment by robofant** [comment]
I just compared 1.1.3 and 1.1.4 and changed back one little thing. It worked for me.
I tested your fix and can confirm it works.

**Comment by robofant** [comment]
Ok, one thing.
In CoolURI I did: DELETE EVERYTHING AND START AGAIN, then I cleared the frontend cache. And then there were a lot of warnings in the system protocol.

TYPO3 7.6.25:
Core: Error handler (FE): PHP Warning: Illegal string offset 'num' in /html/typo3conf/ext/cooluri/Classes/Integration/CoolUri.php line 552

TYPO3 8.7.11:
Core: Error handler (FE): PHP Warning: Illegal string offset 'num' in /html/typo3/typo3conf/ext/cooluri/Classes/Integration/CoolUri.php line 415
